watertap/unit_models/crystallizer.py

```diff
@@ -0,0 +1,696 @@
+        units_meta = self.config.property_package.get_metadata().get_derived_units
+
+        # Add unit variables
+
+        self.hex_max_temp_increase = Param(
+            initialize=4,
+            units=pyunits.K,
+            doc="Maximum temperature difference between inlet and outlet of a crystallizer heat exchanger.\
+            Lewis et al. suggests 1-2 degC but use 5degC in example; Tavare example used 4 degC.\
+            Default is 4 degC")
+
+        # ====== Crystallizer sizing parameters ================= #
+        self.L_d = Param(
+            initialize=0.5e-3, # From Mersmann et al., Tavare et al. example
+            # bounds=(0.2e-3, 0.6e-3), # Limits for FC crystallizers based on Bermingham et al.
+            units=pyunits.m,
+            doc="Desired median crystal size, m"
+            )
+
+        self.G_rate = Param(
+            initialize=3.7e-8, # From Mersmann et al. for NaCl. Perry has values between 0.5e-8 to 13e-8 for NaCl
+            # bounds=(1e-9, 1e-6), # Based on Mersmann and Kind diagram. 
+            units=pyunits.m/pyunits.s,
+            doc="Crystal growth rate, m/s"
+            )
+
+
+        self.k_param = Param(
+            initialize=0.04,
+            units=pyunits.m/pyunits.s,
+            doc="Sounders-Brown constant, set at 0.04 m/s based on Dutta et al. \
+            Lewis et al suggests 0.024 m/s, while Tavare suggests about 0.06 m/s ")
+
+
+        # ====== Model variables ================= #
+        self.crystallization_yield = Var(
+            solute_set,
+            initialize=0.5,
+            bounds=(1e-8, 1),
+            units=pyunits.dimensionless,
+            doc="Crystallizer solids yield"
+            )
+
+        self.product_volumetric_solids_fraction = Var(
+            initialize=0.25,
+            bounds=(1e-8, 1),
+            units=pyunits.dimensionless,
+            doc = "Volumetric fraction of solids in slurry product (i.e. solids-liquid mixture)."
+            )
+
+        self.temperature_operating = Var(
+            initialize=298.15,
+            bounds=(273, 1000),
+            units=pyunits.K,
+            doc="Crystallizer operating temperature: boiling point of the solution."
+            )
+
+        self.pressure_operating = Var(
+            initialize=1e3,
+            bounds=(0.001, 1e6),
+            units=pyunits.Pa,
+            doc="Operating pressure of the crystallizer."
+            )
+
+        self.magma_density = Var(
+            initialize=250,
+            bounds=(1, 5000),
+            units=pyunits.kg/pyunits.m**3,
+            doc="Magma density, i.e. mass of crystals per unit volume of suspension"
+            )
+
+        self.slurry_density = Var(
+            initialize=1000,
+            bounds=(1, 5000),
+            units=pyunits.kg/pyunits.m**3,
+            doc="Suspension density, i.e. density of solid-liquid mixture before separation"
+            )
```

Comment:
`dens` is the IDAES convention for density, I'd go with `dens_mass_magma` and `dens_mass_slurry`

Reply:
Done.